src/main/java/com/fdmgroup/forex/models/Trade.java

```diff
@@ -0,0 +1,86 @@
+
+package com.fdmgroup.forex.models;
+
+import java.util.*;
+
+import org.hibernate.annotations.CreationTimestamp;
+
+import com.fdmgroup.forex.models.composites.TradeId;
+
+import jakarta.persistence.*;
+
+@Entity
+@Table(name = "trades")
+@IdClass(TradeId.class)
+public class Trade {
```

Comment:
Instead of specifying `updatable = false` for some of the fields, we can use `@Immutable` annotation for the whole entity.

Reply:
Addressed in 8c336cf